activerecord/lib/active_record/connection_adapters/postgresql/oid/array.rb

```diff
@@ -23,7 +23,7 @@ def deserialize(value)
             when ::String
               type_cast_array(@pg_decoder.decode(value), :deserialize)
             when Data
-              deserialize(value.values)
+              type_cast_array(value.values, :deserialize)
```

Comment:
`values` in `Data` is an array of serialized strings. So we need to deserialize each strings.